kuwala/core/neo4j/importer/src/PipelineConnector.py

```diff
@@ -33,40 +42,76 @@ def connect_h3_indexes():
 
     Neo4jConnection.close_connection()
 
+    end_time = time.time()
+
+    print(f'Connected H3 indexes in {round(end_time - start_time)} s')
+
+
+def add_pois(df):
+    query = '''
+        MERGE (p:Poi { id: event.id })
+        WITH event, p
+        MATCH (h:H3Index { h3Index: event.h3Index })
+        MERGE (p)-[:LOCATED_AT]->(h)
+    '''
+
+    Neo4jConnection.spark_send_query(df, query)
+
+
+def connect_osm_pois(df):
+    query = '''
+        MATCH (p:Poi { id: event.id })
+        WITH event, p
+        MATCH (po:PoiOSM { id: event.osmId })
+        MERGE (po)-[:BELONGS_TO]->(p)
+    '''
+
+    Neo4jConnection.spark_send_query(df, query)
+
+
+def connect_google_pois(df):
+    query = '''
+        MATCH (p:Poi { id: event.id })
+        WITH event, p
+        MATCH (pg:PoiGoogle { id: event.googleId })
+        MERGE (pg)-[:BELONGS_TO { confidence: event.confidence }]->(p)
+    '''
+
+    Neo4jConnection.spark_send_query(df, query)
+
 
 # Create one single POI node combining OSM and Google
 def connect_pois(df_osm: DataFrame, df_google: DataFrame):
-    df_osm = df_osm.select('id', 'h3Index').withColumnRenamed('h3Index', 'h3IndexOsm').withColumnRenamed('id', 'osmId')
+    start_time = time.time()
+    df_osm = df_osm.select('id', 'h3_index') \
```

Comment:
[suggestion] just for later, all these hardcoded name mappings are better to be moved to a KEY / Value dictionary that is importable across the board. this way you avoid potential mistyping...but nothing to action now.

Reply:
Makes sense. I opened an issue for that.